src/macros.rs

```diff
@@ -56,8 +60,20 @@ macro_rules! log {
         }
     });
 
+    // log!(logger: my_logger, Level::Info, "a log event")
+    (logger: $logger:expr, $lvl:expr, $($arg:tt)+) => ({
+        $crate::log!(logger: $logger, target: $crate::__private_api::module_path!(), $lvl, $($arg)+)
+    });
+
+    // log!(target: "my_target", Level::Info, "a log event")
+    (target: $target:expr, $lvl:expr, $($arg:tt)+) => ({
+        $crate::log!(logger: $crate::logger(), target: $target, $lvl, $($arg)+)
```

Comment:
The default logger should be a zero sized type, otherwise caller needs pay the cost of passing the logger value.

Reply:
Please check if this is what you mean https://github.com/rust-lang/log/pull/664/commits/2bb47d65a5c6f94bdc1d694bec4824431201e31a